config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_computenetworkfirewallpolicyrules.compute.cnrm.cloud.google.com.yaml

```diff
@@ -0,0 +1,362 @@
+                  - name
+                - not:
+                    anyOf:
+                    - required:
+                      - name
+                    - required:
+                      - namespace
+                  required:
+                  - external
+                properties:
+                  external:
+                    description: 'Allowed value: The `name` field of a `Project` resource.'
+                    type: string
+                  name:
+                    description: 'Name of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names'
+                    type: string
+                  namespace:
+                    description: 'Namespace of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/'
+                    type: string
+                type: object
+              resourceID:
+                description: Immutable. Optional. The priority of the resource. Used
+                  for creation and acquisition. When unset, the value of `metadata.name`
+                  is used as the default.
+                type: string
+              ruleName:
+                description: An optional name for the rule. This field is not a unique
+                  identifier and can be updated.
+                type: string
+              targetSecureTags:
+                description: A list of secure tags that controls which instances the
+                  firewall rule applies to. If <code>targetSecureTag</code> are specified,
+                  then the firewall rule applies only to instances in the VPC network
+                  that have one of those EFFECTIVE secure tags, if all the target_secure_tag
+                  are in INEFFECTIVE state, then this rule will be ignored. <code>targetSecureTag</code>
+                  may not be set at the same time as <code>targetServiceAccounts</code>.
+                  If neither <code>targetServiceAccounts</code> nor <code>targetSecureTag</code>
+                  are specified, the firewall rule applies to all instances on the
+                  specified network. Maximum number of target label tags allowed is
+                  256.
+                items:
+                  properties:
+                    name:
+                      description: Name of the secure tag, created with TagManager's
+                        TagValue API. @pattern tagValues/[0-9]+.
+                      type: string
+                    state:
+                      description: '[Output Only] State of the secure tag, either
+                        `EFFECTIVE` or `INEFFECTIVE`. A secure tag is `INEFFECTIVE`
+                        when it is deleted or its network is deleted.'
+                      type: string
+                  required:
+                  - name
+                  type: object
+                type: array
+              targetServiceAccounts:
```

Comment:
I think this should be a reference field?

@justinsb Based on our current naming convention, it should be `targetServiceAccountsRefs` when it is turned into a reference list, though do you have any concerns about the name?

Reply:
Can a tfField point to a list in resourceReferences? If so, I'm happy to change this to targetServiceAccountsRefs.

Reply:
Yes we support it, e.g. https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/config/servicemappings/certificatemanager.yaml#L49

Reply:
@maqiuyujoyce To be clear, I mean that targetServiceAccounts would be a list or Refs, not that it's a property in a list. If I add the following to the service mapping

```
        - tfField: target_service_accounts
          key: serviceAccountRef
          dscription: |-
            A list of service accounts indicating the sets of instances that are applied with this rule
          gvk:
            kind: IAMServiceAccount
            version: v1beta1
            group: iamserviceaccounts.iam.cnrm.cloud.google.com

```
would I be able to add multiple service account refs?

See https://cloud.google.com/compute/docs/reference/rest/v1/networkFirewallPolicies/addRule